projects/firmware_homework102/src/main.c

```diff
@@ -0,0 +1,60 @@
+#include <stdint.h>
+#include <stdlib.h>
+#include <time.h>
+
+#include "interrupt.h"
+#include "log.h"
+#include "soft_timer.h"
+#include "wait.h"
+
+struct Counters {
+  uint8_t counter_a;
+  uint8_t counter_b;
+};
+
+struct Counters *counter_init() {
+  struct Counters *out = malloc(sizeof(struct Counters));
+  out->counter_a = 0;
+  out->counter_b = 0;
+  return out;
+}
+
+void delay(unsigned milliseconds) {
+  clock_t pause;
+  clock_t start;
+
+  pause = milliseconds * (CLOCKS_PER_SEC / 1000);
+  start = clock();
+  while ((clock() - start) < pause) {
+    continue;
+  }
+}
+
+void increment(SoftTimerId timer_id, void *context) {
```

Comment:
This is a private function, so our [coding standards](https://uwmidsun.atlassian.net/wiki/spaces/ELEC/pages/9306218/Coding+Standards) say that it should be `static` and prefixed with `prv_`. (The `static` makes the function file-local, so that if another file declares a `prv_increment` function, we don't have a linking issue.)